src/rdf4cpp/rdf/IRIFactory.cpp

```diff
@@ -158,19 +168,8 @@ std::string IRIFactory::merge_paths(IRIView::AllParts const &base, std::string_v
 
 std::string IRIFactory::construct(std::string_view scheme, std::optional<std::string_view> auth, std::string_view path,
                                   std::optional<std::string_view> query, std::optional<std::string_view> frag) noexcept {
-    size_t l = scheme.size() + 1 + path.size();
-    if (auth.has_value()) {
-        l += auth->size() + 2;
-    }
-    if (query.has_value()) {
-        l += query->size() + 1;
-    }
-    if (frag.has_value()) {
-        l += frag->size() + 1;
-    }
-
     std::string str;
-    str.reserve(l);
```

Comment:
add docstring explaining the formula.

And why does bit_ceil solve the problem?

Reply:
It avoids the branches and preserves power of two growth semantics. This is our best possible change to not realloc if we don't want branches and don't wanto to randomly guess the size of the additional parts